### ISSUE TITLE
`lineSplit` inefficiency using ByteString `findIndices`

### ISSUE THREAD

**Issue by archaephyrryx** [haskell-streaming/streaming-bytestring#14]
TL;DR: The per-byte-cost of `findIndices` (strict bytestring function) has a huge constant factor, rendering lineSplit inefficient

When looking at examples of various optimizations of Streams, I happened to notice a huge discrepancy between the performance of the same code using `lineSplit 1` versus `lines` (both from Data.ByteString.Streaming.Char8) when run over large input.

I checked the implementation source code for `lineSplit` and noticed that it calls the (strict) ByteString function `findIndices` to locate the *n*th newline character in a given byte sequence to capture everything up to and including that character. However, for `lineSplit 1` as a minimal example, this yields abysmal performance, as the per-byte cost of `findIndices` is ~20x greater than `findIndex`.

The call to `findIndices` should ideally be replaced with a function with lower per-byte overhead, which is a bug against the `bytestring` package. ([CR](https://github.com/haskell/bytestring/issues/269))

**Comment by archaephyrryx** [comment]
This has been resolved by #18
